application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubReference.java

```diff
@@ -0,0 +1,168 @@
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.awt.*;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.List;
+
+/**
+ * GitHub Referencing feature. If someone sends #id of an issue (e.g. #207) the bot replies with an
+ * embed that contains info on the issue/pr
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class GitHubReference extends MessageReceiverAdapter {
+    /**
+     * The pattern used to determine whether a message is referencing an issue
+     */
+    protected static final Pattern ISSUE_REFERENCE_PATTERN = Pattern.compile("#(\\d+)");
+
+    private final Config config;
+
+    /**
+     * The repositories that are searched when looking for an issue
+     */
+    private List<GHRepository> repositories;
+
+    public GitHubReference(Config config) {
+        super(Pattern.compile(config.getGitHubReferenceChannelPattern()));
+
+        this.config = config;
+
+        try {
+            acquireRepositories();
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    /**
+     * Scans all repositories mentioned in the config and adds their object representations using
+     * kohuske's API to a list
+     */
+    private void acquireRepositories() throws IOException {
+        repositories = new ArrayList<>();
+
+        GitHub githubApi = GitHub.connectUsingOAuth(config.getGitHubApiKey());
+
+        for (long repo : config.getGitHubRepositories()) {
+            repositories.add(githubApi.getRepositoryById(repo));
+        }
+    }
+
+    @Override
+    public void onMessageReceived(MessageReceivedEvent event) {
+        if (event.getAuthor().isBot()) {
+            return;
+        }
+
+        Message message = event.getMessage();
+        String content = message.getContentRaw();
+        Matcher matcher = ISSUE_REFERENCE_PATTERN.matcher(content);
+        List<MessageEmbed> embeds = new ArrayList<>();
+
+        while (matcher.find()) {
+            int id = Integer.parseInt(matcher.group(1));
+
+            try {
+                Optional<GHIssue> issue = findIssue(id);
+
+                if (issue.isPresent()) {
+                    embeds.add(generateReply(issue.get()));
+                }
+            } catch (IOException ex) {
+                throw new UncheckedIOException(ex);
+            }
+        }
+
+        List<List<MessageEmbed>> partition = ListUtils.partition(embeds, 10);
+        boolean first = true;
+        TextChannel textChannel = message.getChannel().asTextChannel();
+
+        for (List<MessageEmbed> messageEmbeds : partition) {
```

Comment:
move that var down to where its used

Reply:
which var?

Reply:
good question. seems the comment is put at the wrong place. ill recheck

Reply:
ah. it was about the `TextChannel`

Reply:
the else is called every loop iteration except the first

Reply:
ah. fair then. maybe move it right next to the loop then:
```java
Foo foo = ...
for (...) {
  ...
}
```
but at this point its just nitpicking

Reply:
it's already right above the for loop though?

Reply:
ah. fair